### ISSUE TITLE
Q: What is the meaning of numbers at the end of each reaction in datesets?

### ISSUE THREAD

**Issue by zhoujuncsu** [connorcoley/rexgen_direct#2]
What do the bold numbers below mean?
[CH2:15]([CH:16]([CH3:17])[CH3:18])[Mg+:19].[CH2:20]1[O:21][CH2:22][CH2:23][CH2:24]1.[Cl-:14].[OH:1][c:2]1[n:3][cH:4][c:5]([C:6](=[O:7])[N:8]([O:9][CH3:10])[CH3:11])[cH:12][cH:13]1>>[OH:1][c:2]1[n:3][cH:4][c:5]([C:6](=[O:7])[CH2:15][CH:16]([CH3:17])[CH3:18])[cH:12][cH:13]1 **6-8-0.0;15-6-1.0;15-19-0.0**

p.s.  Thank you for your great work on organic reaction AI.

**Comment by connorcoley** [comment]
This almost looks like the first line of the training data file, so I'll reproduce the full one here:

```
[CH2:15]([CH:16]([CH3:17])[CH3:18])[Mg+:19].[CH2:20]1[O:21][CH2:22][CH2:23][CH2:24]1.[Cl-:14].[OH:1][c:2]1[n:3][cH:4][c:5]([C:6](=[O:7])[N:8]([O:9][CH3:10])[CH3:11])[cH:12][cH:13]1>>[OH:1][c:2]1[n:3][cH:4][c:5]([C:6](=[O:7])[CH2:15][CH:16]([CH3:17])[CH3:18])[cH:12][cH:13]1 6-8-0.0;15-6-1.0;15-19-0.0
```

The second half of the line when splitting by whitespace contains information about what _graph edits_ are required to convert the reactants to the product. This is what the core finder is then trained to predict.

Using this example, we have three semi-colon-delimited edits required:

1. ```6-8-0.0``` means atoms 6 and 8 lose their bond (change to a zeroth-order bond), corresponding to the loss of the amide C-N bond
1. ```15-6-1.0``` means atoms 6 and 15 form a single bond
1. ```15-19-1.0```means atoms 15 and 19 lose their bond, corresponding to the loss of the carbon magnesium bond

**Comment by gnsrivastava** [comment]
Hello @connorcoley
What is the meaning of values in column 3,5,7..... in 

[CH2:23]1[O:24][CH2:25][CH2:26][CH2:27]1.[F:1][c:2]1[c:3]([N+:10](=[O:11])[O-:12])[cH:4][c:5]([F:9])[c:6]([F:8])[cH:7]1.[H-:22].[NH2:13][c:14]1[s:15][cH:16][cH:17][c:18]1[C:19]#[N:20].[Na+:21] | 1 | 2-13-1.0 | 2.196 | 1-2-0.0 | 1.877 | 6-8-0.0 | -1.677 | 10-11-0.0 | -2.619 | 6-13-1.0 | -3.103 | 10-12-0.0 | -3.343 | 19-20-0.0 | -5.339 | 19-20-1.0 | -5.793 | 6-19-1.0 | -6.18 | 13-14-0.0 | -6.351 | 10-11-1.0 | -6.595 | 3-19-1.0 | -6.662 | 18-19-0.0 | -6.896 | 2-19-2.0 | -6.987 | 3-10-0.0 | -7.033 | 11-19-2.0 | -7.341 | 5-9-0.0 | -7.476 | 10-19-2.0 | -7.808 | 2-19-1.0 | -7.914 | 5-19-1.0 | -8.056 | 2-13-2.0 | -8.513 | 6-14-1.0 | -8.671 | 12-19-2.0 | -8.699 | 8-18-1.0 | -8.924 | 6-16-1.0 | -9.005
What do they represent?
Any help would be highly appreciated. 

Thanks
Gopal

**Comment by connorcoley** [comment]
These are graph edits, as the previous comment suggests. 2-13-1.0 means that atoms 2 and 13 form a single bond
